arch/arm/boot/dts/overlays/rotary-encoder-overlay.dts

```diff
@@ -0,0 +1,27 @@
+// Device tree overlay for GPIO connected rotary encoder.
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
+
```

Comment:
This overlay looks technically correct, but I would like to see it use the pinctrl feature to reserve the pins used for GPIOs.

For fairly bogus reasons, the pin control and GPIOs subsystems are independent - it is currently possible to claim pin(GPIO) 4 for GPIO and for one of the alternate functions simultaneously. There is now a STRICT feature that we could enable but we haven't done so yet, so I prefer overlays to claim pins using both mechanisms to avoid accidental collisions. See https://github.com/raspberrypi/linux/pull/2011/files for a recent example.